plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, List, MutableMapping, Optional
+
+import optuna
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    RangeSweep,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+from optuna.samplers import CmaEsSampler, RandomSampler, TPESampler
+
+from .config import DistributionConfig, OptunaConfig
+
+log = logging.getLogger(__name__)
+
+
+def create_optuna_distribution_from_config(config: MutableMapping[str, Any]) -> Any:
```

Comment:
Can we use a tighter type for return?

Reply:
I replace `Any` with `BaseDistribution` in https://github.com/facebookresearch/hydra/pull/1132/commits/10108c6337fef4a5f95ad8e300d42ac42cf6705e. Please note that the new `create_optuna_distribution_from_config` raises `NotImplementedError` when an unsupported parameter type is given. IMO, users can find the cause of errors more easily.